src/scripts/checkheader.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# *
+# * This Source Code Form is subject to the terms of the Mozilla Public
+# * License, v. 2.0. If a copy of the MPL was not distributed with this
+# * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# *
+
+echo "[MPL] Checking source files for MPL headers..."
+
+# Checks for the existance of the MPL header
+MPL_REGEX=".*This Source Code Form is subject to the terms of the Mozilla Public\s{0,3}.{0,4}License, v\. 2\.0\. If a copy of the MPL was not distributed with this\s{0,3}.{0,4}file, You can obtain one at http:\/\/mozilla\.org\/MPL\/2\.0\/\."
```

Comment:
This regex requires that the line is split in at most 2 places (after "Public" and after "this"), and that the beginning-of-line comment syntax is between 0 and 4 characters long. Do we really want to be that specific? It also allows any number of characters before the license starts, which seems wrong to me, although I don't have a good solution for files that need something like /bin/bash at the beginning, so we may need to keep that.

Here's an alternative that can split between any word and allows any number of non-word characters between each word to allow for most comment syntaxes. It just uses the pattern `\W+`, which matches one or more non-word characters, between each word in the license. It also seems to be about twice as fast as yours although both are totally negligible (.016 seconds vs .030 seconds to run the script on my machine).

`.*?This\W+Source\W+Code\W+Form\W+is\W+subject\W+to\W+the\W+terms\W+of\W+the\W+Mozilla\W+Public\W+License,\W+v\.\W+2\.0\.\W+If\W+a\W+copy\W+of\W+the\W+MPL\W+was\W+not\W+distributed\W+with\W+this\W+file,\W+You\W+can\W+obtain\W+one\W+at\W+http:\/\/mozilla\.org\/MPL\/2\.0\/\.`

Reply:
Regex implemented in 709246e. I put the header check in its own script and set Travis to run that; I don't think we can lint in Travis just yet (there are too many failures).